partitioner/src/main/java/io/confluent/connect/storage/partitioner/PartitionerConfig.java

```diff
@@ -78,10 +82,17 @@
   public static final String TIMEZONE_DEFAULT = "";
   public static final String TIMEZONE_DISPLAY = "Timezone";
 
-  // Schema group
-  public static final String SCHEMA_GENERATOR_CLASS_CONFIG = "schema.generator.class";
-  public static final String SCHEMA_GENERATOR_CLASS_DOC = "The schema generator to use with partitioners.";
-  public static final String SCHEMA_GENERATOR_CLASS_DISPLAY = "Schema Generator Class";
+  public static final String TIMESTAMP_EXTRACTOR_CLASS_CONFIG = "timestamp.extractor";
+  public static final String TIMESTAMP_EXTRACTOR_CLASS_DOC =
+      "The extractor used to get timestamp for records when partitioning with ``TimeBasedPartitioner``.";
+  public static final String TIMESTAMP_EXTRACTOR_CLASS_DEFAULT = "Wallclock";
+  public static final String TIMESTAMP_EXTRACTOR_CLASS_DISPLAY = "Timestamp Extractor";
+
+  public static final String TIMESTAMP_FIELD_NAME_CONFIG = "timestamp.field";
+  public static final String TIMESTAMP_FIELD_NAME_DOC =
+      "The record field to be used as timestamp by the timestamp extractor.";
+  public static final String TIMESTAMP_FIELD_NAME_DEFAULT = "Timestamp";
```

Comment:
Again, added as a reasonable tradeoff between "no default" which would make this property unnecessarily mandatory and "empty string" which is practically a useless default. 

I'm fine with any reasonable name. That's something anyone who uses custom timestamp fields will probably change anyways.  I'd go with `timestamp`